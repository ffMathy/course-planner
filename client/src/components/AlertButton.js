import React from "react";
import {
    Button,
    AlertDialog,
    AlertDialogContent,
    AlertDialogOverlay,
    AlertDialogBody,
    AlertDialogFooter,
    AlertDialogHeader,
    IconButton,
} from "@chakra-ui/core";

const AlertButton = ({ itemType, itemName, action, navigateTo, confirmFn }) => {
    const [isOpen, setIsOpen] = React.useState();
    const onClose = () => setIsOpen(false);
    const cancelRef = React.useRef();

    return (
        <>
            <IconButton icon="delete" variantColor="red" size="lg" onClick={() => setIsOpen(true)} />
            <AlertDialog isOpen={isOpen} leastDestructiveRef={cancelRef} onClose={onClose}>
                <AlertDialogOverlay />
                <AlertDialogContent>
                    <AlertDialogHeader fontSize="lg" fontWeight="bold">
                        {action} {itemType}
                    </AlertDialogHeader>

                    <AlertDialogBody>
                        Are you sure you want to {action && action.toLowerCase()} {itemName}? You can't undo this action afterwards.
                    </AlertDialogBody>

                    <AlertDialogFooter>
                        <Button ref={cancelRef} onClick={onClose}>
                            Cancel
                        </Button>
                        <Button
                            variantColor="red"
                            onClick={() => {
                                onClose();
                                confirmFn();
                            }}
                            ml={3}
                        >
                            Delete
                        </Button>
                    </AlertDialogFooter>
                </AlertDialogContent>
            </AlertDialog>
        </>
    );
};

export default AlertButton;
