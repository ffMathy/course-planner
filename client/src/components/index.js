import AppConfiguration from "./AppConfiguration";
import OutlineButton from "./OutlineButton";
import ProgrammeCard from "./ProgrammeCard";
import SearchBar from "./SearchBar";
import ProgrammeRequirementsItem from "./ProgrammeRequirementsItem";
import ConfirmationDialog from "./ConfirmationDialog";

export { AppConfiguration, OutlineButton, ProgrammeCard, SearchBar, ProgrammeRequirementsItem, ConfirmationDialog };
