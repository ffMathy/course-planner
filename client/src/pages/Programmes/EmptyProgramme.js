import React from "react";
import { Flex, Text } from "@chakra-ui/core";
import { colors as c } from "../../colors";

const EmptyProgramme = () => {
    return (
        <Flex width="100%" height="100%" justify="center">
            <Flex align="center" justify="center">
                <Text textAlign="center" fontSize="md" color={c.grey} as="i">
                    Create or Select a Programme to view its information here.
                </Text>
            </Flex>
        </Flex>
    );
};

export default EmptyProgramme;
