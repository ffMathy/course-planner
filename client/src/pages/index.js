import HomePage from "./HomePage";
import CoursesPage from "./CoursesPage";
import ProgrammesPage from "./Programmes/ProgrammesPage";
import Students from "./Students";

export { HomePage, CoursesPage, ProgrammesPage, Students };
