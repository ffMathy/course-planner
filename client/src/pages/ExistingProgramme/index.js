import React, { useState } from "react";
import { Flex, Text, Box, IconButton, Stack } from "@chakra-ui/core";
import EditRequirementsBox from "./EditRequirementsBox";
import { OutlineButton, ProgrammeRequirementsItem } from "../../components";

const ExistingProgramme = ({ id }) => {
    const [createNewRequirements, setCreateNewRequirements] = useState(false);
    return (
        <Flex width="100%" direction="column">
            <Flex align="center" justify="center">
                <Flex left="1px" justify="flex-start">
                    {/* TODO: add delete functionality */}
                    <IconButton icon="delete" variantColor="red" right="40px" size="lg" />
                </Flex>
                <Text textAlign="center" fontSize="4xl" color="#0F487E">
                    Programme Requirements
                </Text>
                <Flex right="1px" justify="flex-start">
                    {/* TODO: add edit functionality */}
                    <IconButton icon="edit" variantColor="blue" left="40px" size="lg" />
                </Flex>
            </Flex>
            <Flex align="center" justify="center" marginTop="10px">
                <Text textAlign="center" fontSize="md" color="#000000" as="i">
                    Programme Name
                </Text>
            </Flex>
            <Flex align="center" justify="center" marginTop="10px">
                <Text textAlign="center" fontSize="30px" color="#000000" as="b">
                    {/* TODO: dynamically fetch programme name */}
                    BE (Hons) SE
                </Text>
            </Flex>

            <Flex align="center" justify="center" direction="column">
                <Text textAlign="center" fontSize="md" color="#000000" as="i">
                    Requirements
                </Text>
                <Box bg="#e2e2e2" width="75%" textAlign="center" padding="5px" marginTop="10px">
                    <IconButton width="90%" icon="add" bg="#616161" color="white" onClick={() => setCreateNewRequirements(true)} />
                    <Stack direction="column" height="150px" width="100%" overflowY="scroll" align="center">
                        <ProgrammeRequirementsItem onClick={() => setCreateNewRequirements(true)} />
                        <ProgrammeRequirementsItem onClick={() => setCreateNewRequirements(true)} />
                        <ProgrammeRequirementsItem onClick={() => setCreateNewRequirements(true)} />
                        <ProgrammeRequirementsItem onClick={() => setCreateNewRequirements(true)} />
                        <ProgrammeRequirementsItem onClick={() => setCreateNewRequirements(true)} />
                    </Stack>
                </Box>
            </Flex>
            {createNewRequirements && <EditRequirementsBox closeEdit={() => setCreateNewRequirements(false)} />}
            <Flex justify="center" width="100%">
                <OutlineButton text="Define Template" to={`/programmes/${id}/create-template`} height="60px" />
            </Flex>
        </Flex>
    );
};

export default ExistingProgramme;
