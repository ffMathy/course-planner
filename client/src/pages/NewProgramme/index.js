import React from "react";
import { Flex, Text, Input, Box } from "@chakra-ui/core";
import { OutlineButton } from "../../components";

const NewProgramme = () => {
    const saveProgramme = () => {
        // TODO: save programme here
        console.log("Saved");
    };

    const cancelProgramme = () => {
        // TODO: discard changes here
        console.log("Discarded");
    };

    // TODO: generate ID here
    const id = 0;

    return (
        <Flex width="100%" direction="column">
            <Flex align="center" justify="center">
                <Text textAlign="center" fontSize="4xl" color="#0F487E">
                    Create Programme Requirements
                </Text>
            </Flex>
            <Flex align="center" justify="center" marginTop="10px">
                <Text textAlign="center" fontSize="md" color="#000000" as="i">
                    Programme Name
                </Text>
            </Flex>
            <Flex align="center" justify="center" marginTop="10px">
                <Input variant="flushed" placeholder="Enter Programme Name" textAlign="center" width="50%" size="lg" />
            </Flex>

            <Flex align="center" justify="center" marginTop="50px">
                <Text textAlign="center" fontSize="md" color="#000000" as="i">
                    Requirements
                </Text>
                <Box as="span" ml="2" color="gray.600" bg="gray" />
            </Flex>

            <Flex direction="column" align="center" justify="center" height="100%"></Flex>

            <Flex justify="center" width="100%">
                <OutlineButton text="Define Template" to={`/programmes/${id}/create-template`} height="60px" />
            </Flex>
            <Flex justifyContent="space-evenly" width="100%" align="center">
                <OutlineButton text="Cancel" height="60px" width="200px" onClick={cancelProgramme} />
                <OutlineButton text="Save" height="60px" width="200px" onClick={saveProgramme} />
            </Flex>
        </Flex>
    );
};

export default NewProgramme;
