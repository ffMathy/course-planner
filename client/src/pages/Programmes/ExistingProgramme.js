import React, { useEffect, useState } from "react";
import { Flex, Text, Box, IconButton, Stack, Input } from "@chakra-ui/core";
import { OutlineButton, ProgrammeRequirementsItem } from "../../components";
import EditRequirementsBox from "./EditRequirementsBox";
import { useHistory } from "react-router-dom";
import { AlertButton } from "../../components/";
import useProgrammmes from "./useProgrammes";
import useCourses from "../Courses/useCourses";

const ExistingProgramme = ({ programme, notifyUpdate }) => {
    const history = useHistory();
    const { deleteProgramme, updateProgramme } = useProgrammmes();
    const { data } = useCourses();
    const [programmeDegreeInfo, setProgrammeDegreeInfo] = useState({});
    const [createNewRequirements, setCreateNewRequirements] = useState(false);
    const [openConfirmationDialog, setOpenConfirmationDialog] = useState(false);
    const [editRequirementsHeading, setEditRequirementsHeading] = useState("");
    const [openConfirmationDeleteRegulation, setOpenConfirmationDeleteRegulation] = useState(false);
    const [regulationInEdit, setRegulationInEdit] = useState(null);
    const [isEdited, setIsEdited] = useState(false);
    const [name, setName] = useState("");

    useEffect(() => {
        setProgrammeDegreeInfo(programme);
        setName(programme.name);
    }, [isEdited, programme]);

    const cancelEditProgramme = () => {
        setIsEdited(false);
        setName(programmeDegreeInfo.name);
    };

    const saveEditProgramme = () => {
        const editedProgramme = {
            ...programmeDegreeInfo,
            name: name,
        };

        // TODO: show popup success message
        setIsEdited(false);

        // Update the programme and refresh
        updateProgramme(editedProgramme, notifyUpdate);
    };

    const confirmDelete = () => {
        setOpenConfirmationDialog(false);
        deleteProgramme(programmeDegreeInfo._id, history, notifyUpdate);
    };

    const deleteRegulation = (regulation) => {
        const updatedRegulations = programme.regulations.filter((r) => r._id !== regulation._id);
        const updatedProgramme = { ...programme, regulations: updatedRegulations };
        updateProgramme(updatedProgramme, notifyUpdate);
        setOpenConfirmationDeleteRegulation(false);
        setCreateNewRequirements(false);
    };

    return (
        <Flex width="100%" direction="column">
            <Flex align="center" justify="center">
                <Flex left="1px" justify="flex-start">
                    {/* Deleting the Programme*/}
                    <AlertButton
                        itemName={programmeDegreeInfo.name}
                        itemType={"Programme"}
                        action={"Delete"}
                        isOpen={openConfirmationDialog}
                        confirmFn={() => confirmDelete()}
                    />
                </Flex>
                <Text textAlign="center" fontSize="4xl" color="#0F487E">
                    Programme Requirements
                </Text>
                <Flex right="1px" justify="flex-start">
                    {!isEdited ? (
                        <IconButton icon="edit" variantColor="blue" left="40px" size="lg" onClick={() => setIsEdited(true)} />
                    ) : (
                        ""
                    )}
                </Flex>
            </Flex>
            <Flex align="center" justify="center" marginTop="10px">
                <Text textAlign="center" fontSize="md" color="#000000" as="i">
                    Programme Name
                </Text>
            </Flex>
            <Flex align="center" justify="center" marginTop="10px">
                {isEdited ? (
                    <Input
                        variant="flushed"
                        placeholder="Enter Programme Name"
                        textAlign="center"
                        width="50%"
                        size="lg"
                        value={name}
                        onChange={(event) => setName(event.target.value)}
                    />
                ) : (
                    <Text textAlign="center" fontSize="30px" color="#000000" as="b">
                        {programmeDegreeInfo.name}
                    </Text>
                )}
            </Flex>

            <Flex align="center" justify="center" direction="column">
                <Text textAlign="center" fontSize="md" color="#000000" as="i">
                    Requirements
                </Text>
                <Box bg="#e2e2e2" width="75%" textAlign="center" padding="5px" marginTop="10px">
                    <IconButton
                        width="90%"
                        icon="add"
                        bg="#616161"
                        color="white"
                        onClick={() => {
                            setCreateNewRequirements(true);
                            setRegulationInEdit(null);
                            setEditRequirementsHeading("Creating new requirement");
                        }}
                    />
                    <Stack direction="column" height="150px" width="100%" overflowY="scroll" align="center">
                        {programme.regulations.map((regulation, index) => (
                            <ProgrammeRequirementsItem
                                key={index}
                                itemNumber={`${index + 1}.`}
                                onEdit={() => {
                                    setCreateNewRequirements(true);
                                    setRegulationInEdit(regulation);
                                    setEditRequirementsHeading(`Editing requirement ${index + 1}`);
                                }}
                                pointRequirement={regulation.pointRequirement}
                                points={regulation.points}
                                courseList={data.filter((course) => regulation.courses.includes(course._id))}
                                deleteButton={
                                    <AlertButton
                                        itemName={`regulation ${index + 1}`}
                                        itemType={"Programme Regulation"}
                                        action={"Delete"}
                                        isOpen={openConfirmationDeleteRegulation}
                                        onClose={() => setOpenConfirmationDeleteRegulation(false)}
                                        confirmFn={() => deleteRegulation(regulation)}
                                    />
                                }
                            />
                        ))}
                    </Stack>
                </Box>
            </Flex>
            {createNewRequirements && (
                <EditRequirementsBox
                    heading={editRequirementsHeading}
                    closeEdit={() => setCreateNewRequirements(false)}
                    programme={programme}
                    notifyUpdate={notifyUpdate}
                    regulation={regulationInEdit}
                />
            )}
            <Flex justify="center" width="100%">
                {/* TODO: redefine routing for template plans because they interefere with GET requests with URL */}
                {programmeDegreeInfo.defaultPlan ? (
                    <OutlineButton text="View Template" to={`/programmes/${programmeDegreeInfo._id}/template`} height="60px" />
                ) : (
                    <OutlineButton text="Define Template" to={`/programmes/${programmeDegreeInfo._id}/create-template`} height="60px" />
                )}
            </Flex>

            {/* Save and Close Buttons for Editing */}
            {isEdited ? (
                <Flex justifyContent="space-evenly" width="100%" align="center">
                    <OutlineButton text="Cancel" height="60px" width="200px" onClick={cancelEditProgramme} />
                    <OutlineButton text="Save" height="60px" width="200px" onClick={saveEditProgramme} />
                </Flex>
            ) : (
                ""
            )}
        </Flex>
    );
};

export default ExistingProgramme;
