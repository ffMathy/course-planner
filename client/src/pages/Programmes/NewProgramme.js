import React, { useState, useEffect } from "react";
import { Flex, Text, Input, Box } from "@chakra-ui/core";
import { OutlineButton } from "../../components";
import useProgrammmes from "./useProgrammes";

const NewProgramme = ({ notifyAddition }) => {
    const { createProgramme } = useProgrammmes();
    const [programmeName, setProgrammeName] = useState();
    const [newProgramme, setNewProgramme] = useState({
        name: "",
        regulations: [
            {
                points: 0,
                pointRequirement: "UPTO",
                courses: [],
            },
        ],
        defaultPlan: null,
    });

    useEffect(() => {
        setNewProgramme({
            name: programmeName,
            regulations: [
                {
                    // TODO: add and change regulations here
                    points: 0,
                    pointRequirement: "UPTO",
                    courses: [],
                },
            ],
            // TODO: add default plan here
            defaultPlan: null,
        });
    }, [programmeName]);

    const saveProgramme = () => {
        createProgramme(newProgramme, notifyAddition);
    };

    const cancelProgramme = () => {
        // TODO: discard changes here
        console.log("Discarded");
    };

    // TODO: generate ID here
    const id = 0;

    return (
        <Flex width="100%" direction="column">
            <Flex align="center" justify="center">
                <Text textAlign="center" fontSize="4xl" color="#0F487E">
                    Create Programme Requirements
                </Text>
            </Flex>
            <Flex align="center" justify="center" marginTop="10px">
                <Text textAlign="center" fontSize="md" color="#000000" as="i">
                    Programme Name
                </Text>
            </Flex>
            <Flex align="center" justify="center" marginTop="10px">
                <Input
                    variant="flushed"
                    placeholder="Enter Programme Name"
                    textAlign="center"
                    width="50%"
                    size="lg"
                    onChange={(event) => setProgrammeName(event.target.value)}
                />
            </Flex>

            <Flex align="center" justify="center" marginTop="50px">
                <Text textAlign="center" fontSize="md" color="#000000" as="i">
                    Requirements
                </Text>
                <Box as="span" ml="2" color="gray.600" bg="gray" />
            </Flex>

            <Flex direction="column" align="center" justify="center" height="100%"></Flex>

            <Flex justify="center" width="100%">
                <OutlineButton text="Define Template" to={`/programmes/${id}/create-template`} height="60px" />
            </Flex>
            <Flex justifyContent="space-evenly" width="100%" align="center">
                <OutlineButton text="Cancel" height="60px" width="200px" onClick={cancelProgramme} to={`/programmes/`} />
                <OutlineButton text="Save" height="60px" width="200px" onClick={saveProgramme} to={`/programmes/`} />
            </Flex>
        </Flex>
    );
};

export default NewProgramme;
