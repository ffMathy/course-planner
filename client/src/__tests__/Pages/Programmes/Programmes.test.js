import React from "react";
import renderer from "react-test-renderer";
import { AppConfiguration } from "../../../components";
import { Programmes } from "../../../pages";
import NewProgramme from "../../../pages/Programmes/NewProgramme";
import ExistingProgramme from "../../../pages/Programmes/ExistingProgramme";
import EmptyProgramme from "../../../pages/Programmes/EmptyProgramme";

test("Main Programmes Page", () => {
    const tree = renderer
        .create(
            <AppConfiguration>
                <Programmes />
            </AppConfiguration>,
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("Creating new programmes sub-page ", () => {
    const tree = renderer
        .create(
            <AppConfiguration>
                <NewProgramme />
            </AppConfiguration>,
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("Existing programmes sub-page ", () => {
    let programme = {
        name: "SOFTENG761",
        regulations: [
            {
                points: 15,
                pointRequirement: "ATLEAST",
                courses: [],
            },
        ],
        defaultPlan: null,
    };

    const tree = renderer
        .create(
            <AppConfiguration>
                <ExistingProgramme id={"fakeProgrammeID"} programme={programme} />
            </AppConfiguration>,
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("Empty programmes sub-page ", () => {
    const tree = renderer
        .create(
            <AppConfiguration>
                <EmptyProgramme />
            </AppConfiguration>,
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
