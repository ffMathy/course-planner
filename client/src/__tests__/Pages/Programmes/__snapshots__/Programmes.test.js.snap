// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creating new programmes sub-page  1`] = `
<div
  className="css-1ofqig9"
>
  <div
    className="css-gmuwbf"
  >
    <p
      className="css-19i5rf8"
    >
      Create Programme Requirements
    </p>
  </div>
  <div
    className="css-1yfye7s"
  >
    <i
      className="css-1dbqq24"
    >
      Programme Name
    </i>
  </div>
  <div
    className="css-1yfye7s"
  >
    <input
      className="css-frh740"
      onChange={[Function]}
      placeholder="Enter Programme Name"
    />
  </div>
  <div
    className="css-1ailrle"
  >
    <i
      className="css-1dbqq24"
    >
      Requirements
    </i>
    <span
      className="css-1q0hmy3"
    />
  </div>
  <div
    className="css-16iwlaa"
  />
  <div
    className="css-ldo4d5"
  >
    <a
      href="/programmes/0/create-template"
      onClick={[Function]}
    >
      <button
        className="css-1ad5y8q"
        type="button"
      >
        <p
          className="css-181lpyk"
        >
          Define Template
        </p>
      </button>
    </a>
  </div>
  <div
    className="css-4xdld5"
  >
    <a
      href="/programmes/"
      onClick={[Function]}
    >
      <button
        className="css-1rnlx15"
        onClick={[Function]}
        type="button"
      >
        <p
          className="css-181lpyk"
        >
          Cancel
        </p>
      </button>
    </a>
    <a
      href="/programmes/"
      onClick={[Function]}
    >
      <button
        className="css-1rnlx15"
        onClick={[Function]}
        type="button"
      >
        <p
          className="css-181lpyk"
        >
          Save
        </p>
      </button>
    </a>
  </div>
</div>
`;

exports[`Empty programmes sub-page  1`] = `
<div
  className="css-7eifx2"
>
  <div
    className="css-gmuwbf"
  >
    <i
      className="css-1fwnx19"
    >
      Create or Select a Programme to view its information here.
    </i>
  </div>
</div>
`;

exports[`Existing programmes sub-page  1`] = `
<div
  className="css-1ofqig9"
>
  <div
    className="css-1hyoz7m"
  >
    <p
      className="css-19i5rf8"
    >
      Programme Requirements
    </p>
    <div
      className="css-qzt8sy"
    >
      <div
        className="css-k008qs"
      >
        <button
          aria-controls="menu-5"
          aria-expanded={false}
          aria-haspopup="menu"
          className="css-992tcj"
          id="menubutton-6"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          <svg
            className="css-zgxhu3"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            type="button"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2S13.1 10 12 10zM18 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2S19.1 10 18 10zM6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2S7.1 10 6 10z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
  <div
    className="css-1yfye7s"
  >
    <i
      className="css-1dbqq24"
    >
      Programme Name
    </i>
  </div>
  <div
    className="css-1yfye7s"
  >
    <b
      className="css-1t9wziz"
    >
      SOFTENG761
    </b>
  </div>
  <div
    className="css-u6v8er"
  >
    <i
      className="css-1dbqq24"
    >
      Requirements
    </i>
    <div
      className="css-c0ak17"
    >
      <button
        className="css-1eie9zm"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="css-1im46kq"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
            fill="currentColor"
          />
        </svg>
      </button>
      <div
        className="css-12wbpgd"
      >
        <div
          className="css-1dq1cye"
          onClick={[Function]}
        >
          <div
            className="css-gmuwbf"
          >
            <p
              className="css-7rc2le"
            >
              1.
            </p>
            <p
              className="css-12kv7x2"
            >
              At least
            </p>
            <p
              className="css-12kv7x2"
            >
              15 pts
            </p>
            <p
              className="css-vwuyue"
            >
              FROM
            </p>
            <div
              className="programmeRequirements css-1aqk8yn"
            />
            <button
              className="css-1h1blff"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="css-1im46kq"
                focusable={false}
                role="presentation"
                viewBox="0 0 16 16"
              >
                <path
                  d="M9.41 8l2.29-2.29c.19-.18.3-.43.3-.71a1.003 1.003 0 0 0-1.71-.71L8 6.59l-2.29-2.3a1.003 1.003 0 0 0-1.42 1.42L6.59 8 4.3 10.29c-.19.18-.3.43-.3.71a1.003 1.003 0 0 0 1.71.71L8 9.41l2.29 2.29c.18.19.43.3.71.3a1.003 1.003 0 0 0 .71-1.71L9.41 8z"
                  fill="currentColor"
                  fillRule="evenodd"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-ldo4d5"
  >
    <a
      href="/programmes/undefined/create-template"
      onClick={[Function]}
    >
      <button
        className="css-1ad5y8q"
        type="button"
      >
        <p
          className="css-181lpyk"
        >
          Define Template
        </p>
      </button>
    </a>
  </div>
  
</div>
`;

exports[`Main Programmes Page 1`] = `
<div
  className="css-qxc4md"
>
  <div
    className="css-r7xox5"
  >
    <div
      className="css-1u9fuhx"
    >
      <button
        aria-controls="menu-1"
        aria-expanded={false}
        aria-haspopup="menu"
        className="css-wk9st5"
        id="menubutton-2"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button"
      >
        <svg
          aria-hidden={true}
          className="css-thzv76"
          fill="currentColor"
          focusable="false"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
          />
        </svg>
      </button>
    </div>
    <div
      className="css-9jtr4b"
    >
      <p
        className="css-wqr37p"
      >
        Programmes
      </p>
    </div>
    <div
      className="css-9jn3o8"
    >
      <div
        className="css-yh0f2e"
      >
        <div
          className="css-1lads1q"
        >
          <div
            className="css-3lf2mu"
          >
            <div
              className="css-k6882c"
            >
              <svg
                className="css-1xb1rov"
                focusable={false}
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <input
              className="css-n5gi9c"
              onChange={[Function]}
              placeholder="Search Programmes"
              type="search"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-1nox3ec"
    >
      <div
        className="css-1d3az10"
      >
        <a
          href="/programmes/new"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          <button
            className="css-8nc45l"
            type="button"
          >
            <p
              className="css-1li4w8y"
            >
              <svg
                className="css-1xb1rov"
                focusable={false}
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                  fill="currentColor"
                />
              </svg>
            </p>
          </button>
        </a>
      </div>
    </div>
  </div>
  <div
    className="css-6hu3eg"
  >
    <div
      className="css-trwzgo"
    >
      <div
        className="css-7eifx2"
      >
        <div
          className="css-gmuwbf"
        >
          <i
            className="css-1fwnx19"
          >
            Create or Select a Programme to view its information here.
          </i>
        </div>
      </div>
    </div>
  </div>
</div>
`;
